### PR TITLE
[Mellanox]Add new Mellanox platform device data

### PR DESCRIPTION
**Pull request by keboliu**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?

Some device data is missing for several Mellanox SKU

#### How did you do it?

Add thermal data for ACS-MSN3700 and Mellanox-SN3800-D112C8

Add device data for ACS-MSN4600C and ACS-MSN3420

#### How did you verify/test it?

run related platform test on Mellanox platform

#### Any platform specific information?

This is change is Mellanox platform specific

#### Supported testbed topology if it's a new test case?

N/A

### CODE REVIEW

File: tests/common/mellanox_data.py [hunk 1]
```diff
@@ -547,6 +580,141 @@
...
+            "comex_ambient": {
+                "number": 1
+            }
+        }
+    },
+    "ACS-MSN4600C": {
+        "reboot": {
+            "cold_reboot": True,
+            "fast_reboot": False,
+            "warm_reboot": False
```
Comment (stephenxs) [inline tests/common/mellanox_data.py:618]: Suggest enable warm_reboot for MSN4600C

**Comment by liat-grozovik** [comment]
retest vsimage please

**Comment by keboliu** [comment]
retest vsimage please

**Comment by keboliu** [comment]
retest this please

**Pull request merged**